### PR TITLE
Detach mobile reachability state from non existent webview C-938

### PR DESCRIPTION
**Pull request by nicoback2**
### Description
Screen Shot 2022-09-13 at 12 38 09 PM

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/utils/connectivity.ts [hunk 1]
```diff
@@ -16,4 +19,10 @@ export const checkConnectivity = (netInfo: NetInfoState | null) => {
 export const Connectivity: { netInfo: NetInfoState | null } = { netInfo: null }
 NetInfo.addEventListener((state: NetInfoState) => {
   Connectivity.netInfo = state
+  const newValue = checkConnectivity(state)
```
Comment (dylanjeffers) [inline packages/mobile/src/utils/connectivity.ts:22]: ah very cool

File: packages/web/src/common/store/pages/search-page/lineups/tracks/sagas.js [hunk 2]
```diff
@@ -6,11 +6,12 @@ import {
 } from '@audius/common'
 import { select, all, call, getContext } from 'redux-saga/effects'
 
-import { LineupSagas, isMobile } from 'common/store/lineup/sagas'
+import { LineupSagas } from 'common/store/lineup/sagas'
 import {
   getSearchResults,
   getTagSearchResults
 } from 'common/store/pages/search-page/sagas'
+import { isMobileWeb } from 'common/utils/isMobile'
```
Comment (dylanjeffers) [inline packages/web/src/common/store/pages/search-page/lineups/tracks/sagas.js:14]: i like the rename a lot. should we also change the isMobile file name?

**Review by dylanjeffers — APPROVED** [review]
This all makes sense. a few qs
So with this connectivity util, do we want to make that a "service" kinda like the other stuff, and have a polling daemon type thing in mobile sagas like we do with web sagas? might be overkill just wondering if we can use a similar pattern for both?

Ray was saying something like some phones have the reachability thing built in, but others we need to make a request to one of our services to check. Does that still apply?

File: packages/web/src/common/store/pages/search-page/lineups/tracks/sagas.js (hunk 2 above)
Comment (nicoback2) [reply packages/web/src/common/store/pages/search-page/lineups/tracks/sagas.js:14]: ooh yes!

**Comment by nicoback2** [comment]
@dylanjeffers 

> So with this connectivity util, do we want to make that a "service" kinda like the other stuff, and have a polling daemon type thing in mobile sagas like we do with web sagas? might be overkill just wondering if we can use a similar pattern for both?

Yeah we can definitely do that. So like have a reachability mobile saga and just register the listener there instead of doing it in a side effect?

> Ray was saying something like some phones have the reachability thing built in, but others we need to make a request to one of our services to check. Does that still apply?

I think the netinfo package esp. the latest version is pretty comprehensive in terms of covering all devices + it just pings a URL for devices that don't supply internet connectivity natively https://www.npmjs.com/package/@react-native-community/netinfo

**Comment by dylanjeffers** [comment]
> @dylanjeffers 
> 
> 
> 
> > So with this connectivity util, do we want to make that a "service" kinda like the other stuff, and have a polling daemon type thing in mobile sagas like we do with web sagas? might be overkill just wondering if we can use a similar pattern for both?
> 
> 
> 
> Yeah we can definitely do that. So like have a reachability mobile saga and just register the listener there instead of doing it in a side effect?
> 
> 
> 
> > Ray was saying something like some phones have the reachability thing built in, but others we need to make a request to one of our services to check. Does that still apply?
> 
> 
> 
> I think the netinfo package esp. the latest version is pretty comprehensive in terms of covering all devices + it just pings a URL for devices that don't supply internet connectivity natively https://www.npmjs.com/package/@react-native-community/netinfo

Okay awesome with the NetInfo package. Regarding the saga idea, def up to you on whether it's worth the effort rn vs later or at all :)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--reloaded-reachability

**Review by raymondjacobson — APPROVED** [review]

**Comment by nicoback2** [comment]
> > @dylanjeffers
> > > So with this connectivity util, do we want to make that a "service" kinda like the other stuff, and have a polling daemon type thing in mobile sagas like we do with web sagas? might be overkill just wondering if we can use a similar pattern for both?
> > 
> > 
> > Yeah we can definitely do that. So like have a reachability mobile saga and just register the listener there instead of doing it in a side effect?
> > > Ray was saying something like some phones have the reachability thing built in, but others we need to make a request to one of our services to check. Does that still apply?
> > 
> > 
> > I think the netinfo package esp. the latest version is pretty comprehensive in terms of covering all devices + it just pings a URL for devices that don't supply internet connectivity natively https://www.npmjs.com/package/@react-native-community/netinfo
> 
> Okay awesome with the NetInfo package. Regarding the saga idea, def up to you on whether it's worth the effort rn vs later or at all :)

ok sgg. i started to play around with moving the listener into a saga but then realized that that might be complicating things unnecessarily. so i think we can roll with the current pattern and see if it ever becomes an issue (/Andrew might be reorganizing things anyway throughout the Offline Mode project)

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--reloaded-reachability

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/nkang--reloaded-reachability

**Pull request merged**